pkg/operator/resource/resourceapply/monitoring.go

```diff
@@ -131,11 +128,10 @@ func ApplyUnstructuredResourceImproved(
 	if klog.V(4).Enabled() {
 		klog.Infof("%s %q changes: %v", resourceGVR.String(), namespace+"/"+name, JSONPatchNoError(existing, existingCopy))
 	}
-	var actual *unstructured.Unstructured
-	actual, err = client.Resource(resourceGVR).Namespace(namespace).Update(ctx, existingCopy, metav1.UpdateOptions{})
-	resourcehelper.ReportUpdateEvent(recorder, existingCopy, err)
+	actual, errUpdate := client.Resource(resourceGVR).Namespace(namespace).Update(ctx, existingCopy, metav1.UpdateOptions{})
```

Comment:
why not to `actual, err := ?` do we care if the `err` will be shadowed ?

Reply:
Following the same pattern as [here](https://github.com/openshift/library-go/pull/1823#discussion_r1796709432), I'd prefer to keep this change, but let me know if you think otherwise.